src/workflow/ArcanistDiffWorkflow.php

```diff
@@ -2115,38 +2149,133 @@ private function uberValidateMandatoryFields($message) {
     }
   }
 
-  // runs some validation and returns boolean status if user needs
-  // to edit commit message again
-  private function isCommitEditRequired($message) {
-    if (!$message instanceof ArcanistDifferentialCommitMessage) {
-      return false;
-    }
+  // check and if necessary prompts to enter jira tasks
+  private function attachJiraIssues(
+    ArcanistDifferentialCommitMessage $message = null) {
+
     // atm we do not request automation to add tasks/issues
     if ($this->getArgument('nointeractive')) {
-      return false;
+      return;
+    }
+    // do not execute check for non uber Phabricator though general
+    // recommendation would be to checkout upstream arcanist
+    if (strpos($this->getConduit()->getHost(), "uberinternal.com")===false) {
+      return;
     }
     try {
       phutil_console_require_tty();
-      $config = $this->getConfigurationManager();
-      $check_task_presence = $config->getConfigFromAnySource(
-        'differential.check_task_presence');
-      if (!$check_task_presence) {
-        return false;
-      }
-      $tasks = $message->getFieldValue('maniphestTaskPhids');
-      $issues = $message->getFieldValue('uber-jira.issues');
-      if ($tasks || $issues) {
-        return false;
-      }
-      return phutil_console_confirm(
-        phutil_console_format(
-          '<fg:red>WARNING:</fg> You must associate either Maniphest task or '.
-          'Jira issue with this revision. Do you want to add one?'),
-          $default_no = false);
     } catch (PhutilConsoleStdinNotInteractiveException $e) {
-      // do nothing
+      return;
+    }
+    $config = $this->getConfigurationManager();
```

Comment:
I'd move this custom part to a separate class. Unless there are too many variables which needs to be provided.

Reply:
There are few places which require `$this` I'll move as much as possible and leave minimal amount which requires `ArcanistDiffWorkflow` context